read-fonts/src/tables/postscript/blend.rs

```diff
@@ -0,0 +1,166 @@
+//! Support for the dictionary and charstring blend operator.
+
+use font_types::{BigEndian, F2Dot14, Fixed};
+
+use super::Error;
+use crate::tables::variations::{ItemVariationData, ItemVariationStore, VariationRegionList};
+
+/// The maximum number of region scalars that we precompute.
+const MAX_PRECOMPUTED_SCALARS: usize = 16;
+
+/// State for processing the blend operator for DICTs and charstrings.
+///
+/// To avoid allocation, scalars are computed on demand but this can be
+/// prohibitively expensive in charstrings where blends are applied to
+/// large numbers of elements. To amortize the cost, a fixed number of
+/// precomputed scalars are stored internally and the overflow is computed
+/// as needed.
+///
+/// The `MAX_PRECOMPUTED_SCALARS` constant determines the size of the
+/// internal buffer (currently 16).
+pub struct BlendState<'a> {
+    store: ItemVariationStore<'a>,
+    regions: VariationRegionList<'a>,
```

Comment:
Why is it necessary to store regions? - access should be fast and you only use it to call self.regions.variation_regions.

Perhaps we should not store it (preferred) or store the final result of self.regions.variation_regions?

Reply:
Changed to read regions from IVS when needed.